### PR TITLE
add AsyncZmqThread and AsyncZmqActor classes

### PR DESCRIPTION
**Pull request by jonyrock**
We need some multi threading code. The best way to do it is to use Actor Model. 

This PR contains an implementation of Actor Model based on zmq, asyncio and threads
We haven't choose existing Actor implementation cuz we want zmq. 

## Implementation details

* zqm should handle queue from/to actor. So we should always be able to await next message from actor

* Inheritance is: `Thread <- AsyncZmqThread <- AsyncZmqActor` so it's possible to do `myActor.join()` but please don't do it.

### CODE REVIEW

File: analytics/analytics/utils/concurrent.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    async def put_message(self, message: str):
+        await self.__actor_socket.send_string(message)
+    
+    def __aiter__(self):
+        return self
+    
+    async def __anext__(self) -> str:
+        return await self.__actor_socket.recv_string()
+    
+    # TODO: implement graceful stopping
```
Comment (jonyrock) [inline analytics/analytics/utils/concurrent.py]:
```suggestion
    # TODO: implement graceful stopping

```

File: analytics/analytics/utils/concurrent.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    async def __message_loop(self):
+        while True:
+            text = await self.__zmq_socket.recv_string()
+            asyncio.ensure_future(self._on_message(text))
+
+    async def _send_message(self, message: str):
+        await self.__zmq_socket.send_string(message)
+
+    async def _on_message(self, message: str):
+        """Override this method to receive messages"""
```
Comment (rozetko) [inline analytics/analytics/utils/concurrent.py]: We can wrap it in `@abc.abstractmethod` decorator
https://docs.python.org/3.6/library/abc.html#abc.abstractmethod

File: analytics/analytics/utils/concurrent.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    * async _run(self) -- required
+    * async def _on_message(self, message: str) -- not required
+
+    then run asyncZmqActor.start()
+
+    Example:
+
+    >>>
+    class MyActor(AsyncZmqActor):
+        def async _run(self):
```
Comment (rozetko) [inline analytics/analytics/utils/concurrent.py]: `def async` -> `async def`

File: analytics/analytics/utils/concurrent.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        self.__asyncio_loop.run_until_complete(self._run())
+
+    # TODO: implement stop signal handling
+
+
+class AsyncZmqActor(AsyncZmqThread):
+    """Threaded and Async Actor model based on ZMQ inproc communication
+
+    override following
+    * async _run(self) -- required
```
Comment (amper43) [inline analytics/analytics/utils/concurrent.py]:
```suggestion
    * async def _run(self) -- required
```

File: analytics/analytics/utils/concurrent.py [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    * async def _run(self) -- required
+    * async def _on_message(self, message: str) -- not required
+
+    then run asyncZmqActor.start()
+
+    Example:
+
+    >>>
+    class MyActor(AsyncZmqActor):
+        def async _run(self):
```
Comment (jonyrock) [inline analytics/analytics/utils/concurrent.py]:
```suggestion
        async def _run(self):
```

**Pull request merged**